shell/src-app/Linter.hs

```diff
@@ -0,0 +1,83 @@
+-- Allow ourselves to write Text literals.
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+    import Prologue hiding (Symbol)
+    import Luna.Syntax.Text.Lexer.Runner
+    import Luna.Syntax.Text.Lexer.Token as T
+    import Luna.Syntax.Text.Lexer.Symbol as S
+    import Data.Text.Position            (Delta)
+    import Data.Text32 hiding (convert)
+    import Data.Aeson
+    import qualified Data.ByteString.Lazy as B
+
+    instance ToJSON Delta where
+        toJSON delta = toJSON ((convert delta) :: Int)
+
+    jsonStr :: String -> Value
+    jsonStr = toJSON
```

Comment:
such alias does not have any sense. It introduces another variable to keep in memory and nothing more.

Reply:
Not exactly, there might be another way that I don't know of, but there are multiple possibilities for toJSON (as string literal can be a String, Text, ... and many more) and having this alias with a concrete type removes this ambiguity.

Otherwise I had to type `toJSON ("text" :: String)` everywhere, and now I have jsonStr "text".
If there's another way to avoid specifying the type explicitly all the time, I'll gladly use it :)

Reply:
How often are you converting string literal to JSON?

Reply:
I used to do this a lot, but after the refactor it will be greatly reduced.
For now it's actually in 7 places, to have keywords in lowercase, like this:
```
    valueForSymbol KwCase = Just $ jsonStr "case"
    valueForSymbol KwClass = Just $ jsonStr "class"
    valueForSymbol KwDef = Just $ jsonStr "def"
    valueForSymbol KwForeign = Just $ jsonStr "foreign"
    valueForSymbol KwImport = Just $ jsonStr "import"
    valueForSymbol KwNative = Just $ jsonStr "native"
    valueForSymbol KwOf = Just $ jsonStr "of"
```

Reply:
I suspect that after full refactoirng you will not have literals in any place (especially after auto deriving of instances). Anyway, if you really would need to use monmorphic strings in a module, use `{-# LANGUAGE NoOverloadedStrings #-}` in the header.

Reply:
But I'm actially using `{-# LANGUAGE OverloadedStrings #-}` because aeson wants Text literals which are used for names of fields.

Reply:
Ok, then just type the Strings explicitly in this place, its better than introducing this alias. Anyway, you would not need any of those if used deriving mechanism :D